python/composio/tools/toolset.py

```diff
@@ -628,6 +650,22 @@ def _process_schema_properties(self, action: Action, properties: t.Dict) -> t.Di
             type_="schema",
         )
 
+    def _merge_processors(self, processors: ProcessorsType) -> None:
```

Comment:
The `_merge_processors` method modifies the `self._processors` dictionary in-place. Consider returning a new dictionary instead of modifying the existing one to maintain immutability and prevent potential side effects.